### PR TITLE
Add NetBindable to ScriptComponent

### PR DESCRIPTION
**Pull request by alexeykostin**
## What does this PR do?

Add NetBindable to ScriptComponent to optionally revert LY behavior.

## How was this PR tested?

Build locally

### CODE REVIEW

File: Code/Framework/AzFramework/AzFramework/Script/ScriptComponent.cpp [hunk 1]
```diff
@@ -963,6 +978,63 @@ namespace AzFramework
...
+    {
+        if (m_netBindingTable)
+        {
+            m_netBindingTable->UnbindFromNetwork();
+        }
+    }
+
+    //=========================================================================
+    // CreateNetworkBindingTable
+    // [6/27/2016]
```
Comment (vmednonogov) [inline Code/Framework/AzFramework/AzFramework/Script/ScriptComponent.cpp:1033]: Is this line with data necessary?

File: Code/Framework/AzFramework/AzFramework/Script/ScriptComponent.h [hunk 2]
```diff
@@ -130,5 +174,12 @@ namespace AzFramework
         AZ::Data::Asset<AZ::ScriptAsset>    m_script;               ///< Reference to the script asset used for this component.
         int                                 m_table;                ///< Cached table index
         ScriptPropertyGroup                 m_properties;           ///< List with all properties that were tweaked in the editor and should override values in the m_sourceScriptName class inside m_script.
+
+        // carbonated begin (akostin/mp226-2): Add NetBindable to ScriptComponent
+#if defined(CARBONATED)
+        ScriptNetBindingTable* m_netBindingTable; ///< Table that will hold our networked script values, and manage callbacks
```
Comment (vmednonogov) [inline Code/Framework/AzFramework/AzFramework/Script/ScriptComponent.h:180]: This variable is not a static.
Could you delete it in the destructor?
And maybe protect it from the secondary initialization after delete, if it is necessary.

Comment (alexeykostin) [reply Code/Framework/AzFramework/AzFramework/Script/ScriptComponent.h:180]: Thanks!

File: Code/Framework/AzFramework/AzFramework/Script/ScriptComponent.cpp (hunk 1 above)
Comment (alexeykostin) [reply Code/Framework/AzFramework/AzFramework/Script/ScriptComponent.cpp:1033]: It is was copied from the original source, lets keep it.

File: Code/Framework/AzFramework/AzFramework/Script/ScriptComponent.h (hunk 2 above)
Comment (vmednonogov) [reply Code/Framework/AzFramework/AzFramework/Script/ScriptComponent.h:180]: Another thing:
 m_netBindingTable should be set to NULL in the coonstructor

**Review by vmednonogov — APPROVED** [review]
Approved with a significant note:
m_netBindingTable should be set to NULL in the coonstructor

**Pull request merged**